contracts/allocators/alchemistAllocator.sol

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.7.5;
+
+import "../libraries/Address.sol";
+import "../libraries/SafeMath.sol";
+import "../libraries/SafeERC20.sol";
+
+import "../interfaces/IERC20.sol";
+import "../interfaces/IERC20Metadata.sol";
+import "../interfaces/ITreasury.sol";
+
+import "../types/Ownable.sol";
+
+interface ITokemak_Manager {
+    function currentCycleIndex() external view returns (uint256);
+}
+
+interface ITokemak_tALCX {
+    function deposit(uint256 amount) external;
+
+    function requestedWithdrawals(address addr) external view returns (uint256, uint256);
+
+    function withdraw(uint256 requestedAmount) external;
+
+    function requestWithdrawal(uint256 amount) external;
+}
+
+interface IStakingPools {
+    function claim(uint256 _poolId) external;
+
+    function exit(uint256 _poolId) external;
+
+    function getStakeTotalDeposited(address _account, uint256 _poolId) external view returns (uint256);
+
+    function getStakeTotalUnclaimed(address _account, uint256 _poolId) external view returns (uint256);
+
+    function deposit(uint256 _poolId, uint256 _depositAmount) external;
+
+    function withdraw(uint256 _poolId, uint256 _withdrawAmount) external;
+}
+
+/**
+ *  Contract deploys Alchemist from treasury into the Tokemak tALCX pool,
+ *  tALCX contract gives tALCX token in ratio 1:1 of Alchemist token deposited,
+ *  Contract stake tALCX token on Alchemist staking pool and earn ALCX as reward,
+ *  Contract claims reward and compound it,
+ *  Contract withdraw funds from Alchemist staking pool and Tokemak tALCX pool,
+ *  Sends back Alchemist token with accured reward to treasury.
+ */
+
+contract AlchemistAllocator is Ownable {
```

Comment:
Instead of using Ownable would use OlympusAccessControlled

Reply:
Updated